---
title: "160819 - Load Published NHP data for Blood Manuscript"
author: "Scott Furlan"
output: html_document
---


```{r, echo=FALSE, results='hide'}
remove(list=ls());
library(devtools)
library(dendextend)
library(Vennerable)
library(probedeeper)
library(XLConnect)
library(simpleaffy)
library(xtable)
library(sva)
library(made4)
library(ellipse)
library(ggplot2)
library(gplots)

###LOAD NHP DATA####Load Data
# -------------------------------------
# specify key
# -------------------------------------
date<-"20160819"
key<-""
# -------------------------------------

# -------------------------------------
# find relative directory on Server
# -------------------------------------
setwd("~")
ROOT_DIR<-"~/Dropbox (Kean Lab)/AWS/Scott"
# -------------------------------------
# specify paths and load functions
# -------------------------------------
DATA_DIR <- paste(ROOT_DIR, "/FinalAnalysis/Published/Scripts_BTGVHD/data", sep="") # SPECIFY HERE
DATA_DIR2 <- "~/Dropbox (Kean Lab)/LK NHP/Microarray/Data Files/CEL Files"
PROG_DIR <- paste(ROOT_DIR, "/FinalAnalysis/Published/Scripts_BTGVHD/prog", sep="")      # SPECIFY HERE
RES_DIR  <- paste(ROOT_DIR, "/FinalAnalysis/Published/Scripts_BTGVHD/res", sep="")      # SPECIFY HERE
source(file.path(PROG_DIR, "SFFunc.R"))
finaleset.nhp<-readRDS(file.path(DATA_DIR, "finaleset"))
unfilteredeset.nhp<-readRDS(file.path(DATA_DIR, "unfilteredeset"))

# -------------------------------------
# load color NHP
# -------------------------------------
color.wb<-loadWorkbook(file.path(DATA_DIR, "BreakthroughColor.xlsx"))
color.ws<-readWorksheet(color.wb, "BreakthroughColor", header=TRUE)
ColObj.nhp<-new("ColObj", assign=color.ws, classvec=as.factor(finaleset.nhp$NetworkGroup))
ColObj.nhp<-ColObjInit(ColObj.nhp, LD=70)
```

##Check Assignments
```{r, echo=F}
animal.table<-data.frame(ID=paste(finaleset.nhp@phenoData@data$ID, " Day-", finaleset.nhp@phenoData@data$Day, sep=""), Group=ColObj.nhp@classvec, ArrayTimePoint=finaleset.nhp@phenoData@data$Group, stringsAsFactors=F)
animal.table.split<-split(animal.table$ID, as.factor(animal.table$Group))
animal.table.split.df<-as.data.frame(t(do.call(rbind, lapply(animal.table.split, '[', 1:max(sapply(animal.table.split, length))))))
tli.table <- xtable(animal.table.split.df)
```

```{r, echo=FALSE, results='asis'}
print(tli.table, type="html", include.rownames=FALSE, booktabs=TRUE, floating=FALSE, scalebox=0.6)
```


###LIMMA
```{r, echo=FALSE}
PD<-new("PDObj", eset=unfilteredeset, ColObj=ColObj.nhp, LimmaObj=LimmaObjCreate(unfilteredeset, ColObj.nhp))
```

##Figure 2C Colors
```{r}
step1<-40
step2<-40
line<-vector()
fill<-vector()
line[1]<-"#931a1d"
fill[1]<-LightenDarkenColor(line[1], step1)
line[2]<-LightenDarkenColor(line[1], step2)
fill[2]<-LightenDarkenColor(line[2], step1)
line[3]<-LightenDarkenColor(line[2], step2)
fill[3]<-LightenDarkenColor(line[3], step1)
line[4]<-LightenDarkenColor(line[3], step2)
fill[4]<-LightenDarkenColor(line[4], step1)
pie(rep(1,4), col=fill, labels=fill)
pie(rep(1,4), col=line, labels=line)

step1<-40
step2<-30
lineb<-vector()
fillb<-vector()
start<-"#F7931D"
lineb[1]<-LightenDarkenColor(start, -50)
fillb[1]<-LightenDarkenColor(lineb[1], step1)
lineb[2]<-LightenDarkenColor(lineb[1], step2)
fillb[2]<-LightenDarkenColor(lineb[2], step1)
lineb[3]<-LightenDarkenColor(lineb[2], step2)
fillb[3]<-LightenDarkenColor(lineb[3], step1)
lineb[4]<-LightenDarkenColor(lineb[3], step2)
fillb[4]<-LightenDarkenColor(lineb[4], step1)
pie(rep(1,4), col=fillb, , labels=fillb)
pie(rep(1,4), col=lineb, , labels=lineb)
```


#Figure 3A
``` {r Survival, echo=FALSE}
sel<-ColObj.nhp@classvec=="Breakthrough" | ColObj.nhp@classvec=="Primary"
pcadata<-t(exprs(finaleset.nhp)[,sel])

rownames(pcadata)<-make.unique(rownames(pcadata))
pca1 <- dudi.pca(pcadata, scann = FALSE, nf=10)
s.class(dfxy = pca1$li, fac = ColObj.nhp@classvec[sel], xax = 1, yax = 3)
```

#Figure 3B
```{r}
pcadata<-t(exprs(finaleset.nhp))
#pcadata<-t(data.sel.nhp)
rownames(pcadata)<-make.unique(rownames(pcadata))
pca1 <- dudi.pca(pcadata, scann = FALSE, nf=10)

###PCA SURV###
selectionindex<-vector()
selectionindex[!is.na(suppressWarnings(as.numeric(finaleset.nhp$Survival)))]<-"TRUE"
selectionindex[ColObj.nhp@classvec=="Controlled"]<-"FALSE"
selectionindex[is.na(selectionindex)]<-"FALSE"
selectionindex<-as.logical(selectionindex)
Survival.sel<-as.numeric(finaleset.nhp$Survival[selectionindex])
lm.df<-data.frame(PC1=as.numeric(pca1$li$Axis1[selectionindex]), Survival=as.numeric(log(Survival.sel)), Group=as.factor(as.character(ColObj.nhp@classvec[selectionindex])))
reg1<-(lm(PC1~Survival, lm.df))
summary(reg1)
anova(reg1)
lm.df.melt<-melt(lm.df)
g<-ggplot()+
  geom_point(data=lm.df, aes(x=Survival, y=PC1), col=ColObj.nhp@full$line[selectionindex], size=3)+
  geom_smooth(data=lm.df, aes(x=Survival, y=PC1), method=lm)+
  theme_bw()+
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())+
  theme(axis.ticks.y = element_blank(), axis.text.y = element_blank())+
  labs(x="log Survival", y="PC1")
print(g)
```




#Figure 1C
```{r}
library(survival)
survobj<-Surv(Survival.sel)
survfit<-survfit(survobj~lm.df$Group)
plot(survfit, col=ColObj.nhp@match$line[c(2,5)])
print(survfit)
```

#Make GSEA Files
```{r}
###GSEA Data###
data.sel.nhp<-exprs(finaleset)
classvec.sel.nhp<-ColObj.nhp@classvec
data.gct.nhp<-data.sel.nhp[,order(classvec.sel.nhp)]
colnames(data.gct.nhp)<-make.unique(as.character(classvec.sel.nhp)[order(classvec.sel.nhp)])
gsea.write.gct(data.gct.nhp, file.path(DATA_DIR, "NHP_Allnetworkmodules.gct"))
classvec.sel.cls.nhp<-classvec.sel.nhp[order(classvec.sel.nhp)]
gsea.write.cls(as.factor(classvec.sel.cls.nhp), file.path(DATA_DIR, "NHP_Allnetworkmodules.cls"))
GSEA.ReadClsFile(file.path(DATA_DIR, "NHP_Allnetworkmodules.cls"))
```

#Figure 3C + 4A
####ssGSEA data were generated using the GenepPattern Server using the following input files
1. NHP_Allnetworkmodules.gct
2. NHP_Allnetworkmodules.cls

####Output file is ssGSEA_masterDB_nodupes.gct

####The ConstellationMapping code was run on GenePattern Server and had output details summarized in the following textfiles
1. PrimaryUpConstellationMap.txt
2. BreakthroughUpConstellationMap.txt
####Input files included
1. ssGSEA_masterDB_nodupes.gct
2. NHP_Allnetworkmodules.cls


#Figure 3D
```{r}
Comparison1<-"Primary"
Comparison2<-c("Healthy.Control", "Breakthrough")
masterDB<-readGMT.SF(file.path(DATA_DIR, "MSigDBALL.gmt"))
toGMT<-masterDB[grep("GSE14308",names(masterDB))]
MCGO3<-MultipleClassGSEA(data.gct.nhp,
                         Comparison1, Comparison2, toGMT, 
                         classvec=as.factor(classvec.sel.cls.nhp),
                         runGSEAcode=TRUE, reshuffling.type="gene.labels",
                         uniquelabel="GSE14308-rearranged-PvOthers",
                         directory=DATA_DIR)
```

#Figure4C
```{r}
MAboxplot("MKI67", PD, reorder=c(4,1,5,3,2))
MAboxplot("GZMB", PD, reorder=c(4,1,5,3,2))
```


##Figure 4C Pivot Transcripts
```{r}
limmaObj<- LimmaObjCreate(finaleset, ColObj.nhp, lfc.thresh=0.5)
slotNames(limmaObj)
str(limmaObj@Contrasts)
limmaObj@Contrasts$meaning
limmaObj2<- LimmaObjCreate(unfilteredeset, ColObj.nhp, lfc.thresh=0.5)

Symbol<-rownames(limmaObj@DEGenes[[14]])[rownames(limmaObj@DEGenes[[14]]) %in% rownames(limmaObj@DEGenes[[16]])]
colnames(limmaObj@DEGenes[[14]])
X<-limmaObj@DEGenes[[14]][Symbol,]$logFC
Y<-limmaObj@DEGenes[[16]][Symbol,]$logFC
t<--log(limmaObj@DEGenes[[14]][Symbol,]$adj.P.Val)
df<-data.frame(Symbol=Symbol, X=X, Y=Y, t=t)
ggplot(df)+
  geom_point(aes(x=X, y=Y, size=t))

Symbol<-rownames(limmaObj@AllGenes[[14]])[rownames(limmaObj@AllGenes[[14]]) %in% rownames(limmaObj@AllGenes[[16]])]
colnames(limmaObj@AllGenes[[14]])
X<-limmaObj@AllGenes[[14]][Symbol,]$logFC
Y<-limmaObj@AllGenes[[16]][Symbol,]$logFC
t<--log(limmaObj@AllGenes[[14]][Symbol,]$adj.P.Val)
df2<-data.frame(Symbol=Symbol, X=X, Y=Y, t=t)
Color<-vector()
Color[which(df2$Symbol %in% df$Symbol)]<-"red"
Color[which(!df2$Symbol %in% df$Symbol)]<-"black"
df2$Color<-Color

BTUP<-as.character(df$Symbol[intersect(which(df$X>0), which(df$Y<0))])
PUP<-as.character(df$Symbol[intersect(which(df$X<0), which(df$Y>0))])
df3<-df[df$Symbol %in% c(BTUP, PUP),]


ggplot(df2)+
  geom_point(aes(x=X, y=Y, size=t), colour="black", alpha=0.1)+
  geom_point(data=df, aes(x=X, y=Y, size=t), colour="blue", alpha=0.2)+
  geom_point(data=df3, aes(x=X, y=Y, size=t), colour="red")+
  geom_text(data=df3, aes(x=X, y=Y, label=Symbol), hjust=0, vjust=0, colour="red")+
  theme_bw()
```


##Figure 4D Hill Heatmap
###1. DE list of Breakthrough vs Auto vs transcripts listed in the Hill paper.
``` {r, echo=FALSE}
#Hill
con<-file(file.path(DATA_DIR, "Hill.txt"))
HILL_UP<-toupper(unlist(strsplit(readLines(con, warn="FALSE")[1], " ")))
HILL_DOWN<-toupper(unlist(strsplit(readLines(con, warn="FALSE")[2], " ")))
close(con)
Hill<-c(HILL_UP, HILL_DOWN)

df<-data.frame(Gene=rownames(PD@LimmaObj@AllGenes[[1]][Hill,]), BvA=PD@LimmaObj@AllGenes[[1]][Hill,1], BvHC=PD@LimmaObj@AllGenes[[14]][Hill,1])
mat<-as.matrix(df[,2:3])
rownames(mat)<-df$Gene
dir<-vector()
dir[1:12]<-df$BvHC[1:12]>0
dir[13:24]<-df$BvHC[13:24]<0
names(dir)<-as.character(df$Gene)
dir2<-vector()
dir2[1:12]<-df$BvA[1:12]>0
dir2[13:24]<-df$BvA[13:24]<0
names(dir2)<-as.character(df$Gene)
dir2

dir3<-dir2 | dir
rmat<-t(as.matrix(cbind(c("Black", "Grey")[as.factor(dir3)], RowSide)))
heatmap.3(mat, dendrogram="none", Rowv=FALSE, scale="none", col=gsea.col, RowSideColors=rmat, trace="none", density.info="none", cexCol = 2, main="RowSide by Auto OR HC")
```

#WGCNA - Figures 6A-D
####Load files
```{r}
library(Biobase) #Required for extracting individual classes of the eSet
library(WGCNA)
library(RColorBrewer)   ###used to create nicer colour palettes
library(preprocessCore) ###used by the quantile normalization function
library(RColorBrewer)   ###used to create nicer colour palettes
library(preprocessCore) ###used by the quantile normalization function
library(reshape)
library(doParallel)
numCores<-detectCores(all.tests = TRUE, logical = FALSE)
registerDoParallel(cores=numCores)
if(numCores>1){
  allowWGCNAThreads()
}
options(stringsAsFactors=FALSE)
source(file.path(PROG_DIR, "MasterNetworkFile.R"))
source(file.path(PROG_DIR, "functionList.R"))
data<-finaleset
group = data$NetworkGroup
exprData = exprs(data)
nhpcolors<-ColObj.nhp@match$line
#pie(rep(1, length(nhpcolors)), labels=names(nhpcolors), col=nhpcolors)
colnames(data@phenoData@data)
datTraits<-data@phenoData@data[,c(3,4,5,7:14)]
print(summary(as.factor(group)))
```

##Find `r numVar` most variant genes
```{r variance, echo=TRUE}
numVar<-4000
rowVar<-apply(exprData, 1,var)
rowVar.ord<-(rowVar[order(-rowVar)])
ab<-min(rowVar.ord[1:numVar])
hist(rowVar, breaks=1000, main="Variance of Expression Set and Threshold Used")
abline(v=ab, col="red")
#which(names(rowVar.ord[1:5000]) %in% "IL17A")
exprData.var<-exprData[names(rowVar.ord[1:numVar]),]
```

##Normalize data

```{r Normalize0,echo=TRUE}
#First, center and rescale all samples and features to have mean 0 variance 1
#Note that this 'double centering' means samples are not exactly mean 0.
SFHist(exprData.var, colors=nhpcolors[as.factor(group)], title="Input Data")
```

```{r Normalize1,echo=TRUE}
dataNorm = center.and.scale.by.sample(exprData.var)
SFHist(dataNorm, colors=nhpcolors[as.factor(group)], title="Normalize by Sample")
```

```{r Normalize2,echo=TRUE}
dataNorm2 = center.and.scale.by.feature(dataNorm)
SFHist(dataNorm2, colors=nhpcolors[as.factor(group)], title="Normalize by Feature")
```

##Plot basic dendrogram, including group data. 
```{r firstWGCNADendrogramPlusTraits,echo=TRUE}
#cluster with all samples included
sampleTree = hclust(dist(t(dataNorm2)), method = "average")
# Convert traits to a color representation: white means low, red means high, grey means missing entry
traitColors = numbers2colors(as.numeric(factor(group)), signed = FALSE);
# Plot the sample dendrogram and the colors underneath.
plotDendroAndColors(sampleTree, traitColors,
                    groupLabels = group, cex.dendroLabels = 0.5,
                    main = "Dendrogram with all samples")

```

##First, find soft threshold value for network/module analysis
```{r softThresholdingCalculation,echo=TRUE}
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
sft = pickSoftThreshold(t(dataNorm2[,]), powerVector = powers, verbose = 5)

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=0.9,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
graphics.off()
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"),ylim=c(0,50))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=0.9,col="red")

softPower<-min(sft$fitIndices[sft$fitIndices$SFT.R.sq>0.9,]$Power)
meanCon<-sft$fitIndices[sft$fitIndices$Power==softPower,]$mean.k
medCon<-sft$fitIndices[sft$fitIndices$Power==softPower,]$median.k
abline(v=meanCon,col="red")
```

##Here we create modules of correlated genes in the complete dataset, and ask whether 'eigengenes' are associated with particular traits. 
```{r createTOMTree, echo=TRUE}
#Calculate adjacencies
adjacency = adjacency(t(dataNorm2))

#Calculate TOMs
TOM.all = TOMsimilarity(adjacency)
dissTOM = 1-TOM.all

#Heirarchical clustering on TOM all data
mzTree.all = hclust(as.dist(dissTOM), method = "average");
plot(mzTree.all, xlab="", sub="", main = "Gene clustering on TOM-based dissim", labels = FALSE, hang = 0.04);
```

##And then place them in modules
```{r createModules}
#All data
#Set minimum module size
minModuleSize = meanCon;
# Module identification using dynamic tree cut:
dynamicMods.all = cutreeDynamic(dendro = mzTree.all, distM = dissTOM,
  deepSplit = 3, pamRespectsDendro = FALSE, cutHeight=.99,
  minClusterSize = 2*minModuleSize);
(table(dynamicMods.all))
length(dynamicMods.all)

```

##Now  plot dendrogram and modules
```{r plotDendroAndModules}
# Convert numeric lables into colors
dynamicColors.all = labels2colors(dynamicMods.all)
table(dynamicColors.all)
# Plot the dendrogram and colors underneath
#sizeGrWindow(8,6)
plotDendroAndColors(mzTree.all, dynamicColors.all, "Dynamic Tree Cut",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = "Gene dendrogram and modules")
```

##Merge modules and set up final dataset
```{r mergeModules, echo=TRUE}
# Merge module function, all data
MEList.all = moduleEigengenes(t(dataNorm2), colors = dynamicColors.all)
MEs.all = MEList.all$eigengenes
MEDiss.all = 1-cor(MEs.all);
METree.all = hclust(as.dist(MEDiss.all), method = "average");
MEDissThres.all = 0.25
merge.all = mergeCloseModules(t(dataNorm2), dynamicColors.all, cutHeight = MEDissThres.all, verbose = 3)
plot(METree.all, main = "Clustering of module eigengenes",
xlab = "", sub = "")
abline(h=MEDissThres.all, col = "red")
length(dynamicColors.all)
nrow(dataNorm2)
merge = mergeCloseModules(t(dataNorm2), dynamicColors.all, cutHeight = MEDissThres.all, verbose = 3)
mergedColors = merge$colors;
mergedMEs = merge$newMEs;
plotDendroAndColors(mzTree.all, cbind(dynamicColors.all, mergedColors),
c("Dynamic Tree Cut", "Merged dynamic"),
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)
# Rename to moduleColors
finalmoduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(finalmoduleColors, colorOrder)-1;
MEs = mergedMEs;

```

##Examine module-trait associations
```{r defineMetsAndTraits,echo=TRUE, fig.height=7, fig.width=9, warning=FALSE}

levels(as.factor(finalmoduleColors))
# Define numbers of genes and samples
nMZ.all = ncol(dataNorm2);
nSamples.all = nrow(dataNorm2);
# Recalculate MEs with color labels
MEs0.all = moduleEigengenes(t(dataNorm2), finalmoduleColors)$eigengenes
MEs.all = orderMEs(MEs0.all)
moduleTraitCor = WGCNA::cor(MEs.all, y=datTraits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples.all);

# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
labeledHeatmap(Matrix = moduleTraitCor,
xLabels = names(datTraits),
yLabels = names(MEs.all),
ySymbols = names(MEs.all),
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = TRUE,
cex.text = 0.5,
zlim = c(-1,1),
main = paste("Module-trait relationships"))
plotTOM = dissTOM^7;
# Set diagonal to NA for a nicer plot
diag(plotTOM) = NA;
nSelect = 1000
# For reproducibility, we set the random seed
set.seed(10);
select = sample(numVar, size = nSelect);
selectTOM = dissTOM[select, select];
# There’s no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.
selectTree = hclust(as.dist(selectTOM), method = "average")
selectColors = finalmoduleColors[select];
recoloredColors<-finalmoduleColors
recoloredColors[which(finalmoduleColors %in% "lightyellow")]<-"orange"
recoloredColors[which(finalmoduleColors %in% "cyan")]<-"pink"
recoloredColors[which(finalmoduleColors %in% "turquoise")]<-"darkred"
recoloredColors[which(finalmoduleColors %in% "darkred")]<-"turquoise"
recoloredColors[which(finalmoduleColors %in% "brown")]<-"cyan"
# Open a graphical window
sizeGrWindow(9,9)
# Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing
# the color palette; setting the diagonal to NA also improves the clarity of the plot
subset<-finalmoduleColors %in% c("lightyellow", "black")
sortedTOM<-dissTOM[subset, subset]
plotDiss = selectTOM^7;
plotDiss2 = sortedTOM^7;
diag(plotDiss) = NA;
diag(plotDiss2) = NA;
TOMplot(plotDiss, selectTree, selectColors, main = "Network heatmap plot, selected genes")
heatmap.2(plotDiss2, trace="none", ColSideColors=recoloredColors[subset], RowSideColors=recoloredColors[subset])
# Recalculate module eigengenes
MEs = moduleEigengenes(t(dataNorm2), recoloredColors, excludeGrey=FALSE)$eigengenes
# Isolate weight from the clinical traits
cohort = as.data.frame(datTraits[,7:11]);
names(cohort) = colnames(datTraits[,7:11])
# Add the weight to existing module eigengenes
MET = orderMEs(cbind(MEs, cohort))
# Plot the relationships among the eigengenes and the trait
#sizeGrWindow(5,7.5);
par(cex = 0.9)
plotEigengeneNetworks(MET, "", marDendro = c(0,4,1,2), marHeatmap = c(3,4,1,2), cex.lab = 0.8, xLabelsAngle = 90, excludeGrey=FALSE)
# Plot the dendrogram
sizeGrWindow(6,6);
par(cex = 1.0)
plotEigengeneNetworks(MET, "Eigengene dendrogram", marDendro = c(0,4,2,0), excludeGrey=FALSE, plotHeatmaps = FALSE)
# Plot the heatmap matrix (note: this plot will overwrite the dendrogram plot)
par(cex = 1.0)
plotEigengeneNetworks(MET, "Eigengene adjacency heatmap", marHeatmap = c(3,4,2,2),
plotDendrograms = FALSE, excludeGrey=FALSE, xLabelsAngle = 90)
```

#Supp Fig 3
```{r}
module.list<-split(rownames(dataNorm2), as.factor(recoloredColors))
saveRDS(module.list, file.path(DATA_DIR, "WGCNA_modulelist.rds"))
names(module.list[grep("RORC", module.list)])
gcolors<-nhpcolors[as.factor(group)]
BTgenes<-geneCorPlot(dataNorm2, "RORC", 1e-10, cols=gcolors)
classvec<-group

Other<-c("IL26", "IL1R1", "CCR6", "AHR", "IL23R", "IL22")
for(i in Other){
df<-data.frame(x=dataNorm2["RORC",], y=dataNorm2[i,], group=classvec)
reg1<-(lm(x~y, df))
R2<-summary(reg1)$adj.r.squared
anova(reg1)
PValue<-summary(reg1)$coefficients[2,4]
df.melt<-melt(df, id.vars="group")
colScale <- scale_colour_manual(name = "grp",values = gcolors)
g<-ggplot()+
  geom_point(data=df, aes(x=x, y=y, colour=group), size=1.2)+
  geom_smooth(data=df, aes(x=x, y=y), method=lm, size=0.5, colour="black")+
  theme_bw()+
  #theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())+
  #theme(axis.ticks.y = element_blank(), axis.text.y = element_blank())+
  theme(legend.position="none")+
  labs(x="RORC", y=i)
xlower<-ggplot_build(g)$panel$ranges[[1]]$x.range[2]
ylower<-ggplot_build(g)$panel$ranges[[1]]$y.range[1]
labels.df<-data.frame(x=rep(xlower,2), y=c(ylower, ylower-0.5), text=c(paste("R2=", signif(R2, 3)), paste("P=", signif(PValue, 3))))
g<-g+colScale
g<-g+geom_text(data=labels.df, aes(label = text, x=x, y=y), size=3, vjust = "inward", hjust = "inward")
print(g)
#ggsave(plot=g,height=7,width=7, filename=paste(figpath, "/PCASurv.pdf", sep=""), useDingbats=FALSE)
}
```

##Export to cytoscape - threshold 0.05
```{r export to Cytoscape, echo=TRUE}
# Select modules
modules = c("black");
# Select module probes
probes = rownames(dataNorm2)
inModule = is.finite(match(recoloredColors, modules));
modProbes = probes[inModule];
# Select the corresponding Topological Overlap
modTOM = TOM.all[inModule, inModule]
dimnames(modTOM) = list(modProbes, modProbes)
# Export the network into edge and node list files Cytoscape can read
classvec<-as.factor(data$NetworkGroup)
levels(classvec)
threshold=0.05

cyt <- exportNetwork.SF(modTOM,
edgeFile = paste("CytoscapeInput-edgesSF-", paste(modules, threshold, collapse="-"), ".txt", sep=""),
nodeFile = paste("CytoscapeInput-nodesSF-", paste(modules, threshold, collapse="-"), ".txt", sep=""),
weighted = TRUE,
abbreviated = TRUE,
threshold = threshold,
nodeNames = NULL,
altNodeNames = NULL,
nodeAttr = finalmoduleColors[inModule],
writeDEData=TRUE, expressionset=data, classvec=classvec, comparison=c(5,1), directory=RES_DIR)

# Select modules
modules = c("orange");
# Select module probes
probes = rownames(dataNorm2)
inModule = is.finite(match(recoloredColors, modules));
modProbes = probes[inModule];
# Select the corresponding Topological Overlap
modTOM = TOM.all[inModule, inModule]
dimnames(modTOM) = list(modProbes, modProbes)
# Export the network into edge and node list files Cytoscape can read
classvec<-as.factor(data$NetworkGroup)
levels(classvec)
threshold=0.05

cyt <- exportNetwork.SF(modTOM,
edgeFile = paste("CytoscapeInput-edgesSF-", paste(modules, threshold, collapse="-"), ".txt", sep=""),
nodeFile = paste("CytoscapeInput-nodesSF-", paste(modules, threshold, collapse="-"), ".txt", sep=""),
weighted = TRUE,
abbreviated = TRUE,
threshold = threshold,
nodeNames = NULL,
altNodeNames = NULL,
nodeAttr = finalmoduleColors[inModule],
writeDEData=TRUE, expressionset=data, classvec=classvec, comparison=c(5,1), directory=RES_DIR)
```

#Figure 6F
```{r}
modules = c("black");
# Select module probes
probes = rownames(dataNorm2)
inModule = is.finite(match(recoloredColors, modules));
modProbes = probes[inModule];
BlackPW<-autoDAVID(modProbes)
BlackPW.Sig<-BlackPW[BlackPW$Benjamini< 0.05,]
df<-data.frame(Pathway=BlackPW.Sig$Term[order(BlackPW.Sig$Benjamini)], Significance=BlackPW.Sig$Benjamini[order(BlackPW.Sig$Benjamini)])
mat<-as.matrix(log(df$Significance, 10))
rownames(mat)<-df$Pathway
#mat<-mat[order(mat[,1]),]
rgb.palette <- colorRampPalette(c("red", "yellow"), space = "rgb")
my.padding <- list(layout.heights = list( 
                        top.padding = 0, 
                        main.key.padding = 0, 
                        key.axis.padding = 0, 
                        axis.xlab.padding = 0, 
                        xlab.key.padding = 0, 
                        key.sub.padding = 0), 
                layout.widths = list( 
                        left.padding = 20, 
                        key.ylab.padding = 0, 
                        ylab.axis.padding = 0, 
                        axis.key.padding = 0, 
                        right.padding = 5) 
                ) 
levelplot(mat, scales=list(x=list(rot=45)), col.regions=rgb.palette(120), xlab = 
NULL, ylab = NULL, par.settings = my.padding)
barplot(mat[,1], horiz=TRUE)
dflog<-data.frame(Pathway=BlackPW.Sig$Term, Significance=-log(BlackPW.Sig$Benjamini,10))
ggplot(dflog, aes(y=Significance, x=reorder(Pathway, Significance)))+
  geom_bar( stat="identity")+
  coord_flip()+
        theme_bw() +
      theme(axis.text=element_text(size=16),
          axis.title.x=element_text(size=20, vjust=1),
          axis.text.x = element_text(angle = 45, hjust=1),
          axis.title.y=element_text(size=16, vjust=0.5), plot.title = element_text(vjust = 0, size=20),
          axis.line = element_line(colour = "black"),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.border = element_blank(),
          panel.background = element_blank())
```
#Appendix
```{r Appendix,echo=FALSE}
sessionInfo()
getwd()
```


